app/util/address/index.js

```
@@ -149,6 +149,8 @@ export async function importAccountFromPrivateKey(private_key) {
  * @returns {Boolean} - Returns a boolean
  */
 export function isQRHardwareAccount(address) {
+  if (!isValidHexAddress(address)) return false;
```

We would probably want to throw here since an invalid address is neither true or false WRT being a hardware account.